pymeasure/instruments/advantest/advantestR624X.py

```diff
@@ -0,0 +1,1040 @@
+        2 Wait for sequence program wait time
+        1 Sequence program running
+        0 Synchronous operation state        
+        """,
+        cast=int,
+    )
+
+    error_register = Instrument.measurement(
+        "err?",
+        """ A query command that reads the contents of the error register.
+        """,
+        cast=int,
+    )
+
+    self_test = Instrument.measurement(
+        "*tst?",
+        """ A query command that runs a self-test and reads the result.
+        """,
+        cast=int,
+    )
+
+    optional_equipment = Instrument.measurement(
+        "*opt?",
+        """ A query command that reads the contents of optional equipment.
+        """,
+        cast=int,
+    )
+
+    def set_trigger_link_function(self, enable):
+        """ This command sets the trigger link function on or off.
+        """
+        # 1 OFF
+        # 2 ON
+        enable = int(enable == True) + 1
+
+        self.write(f'tlnk 0,{enable}')
+        self.check_errors()
+
+    def set_display(self, enable):
+        """ This command sets the display on or off.
+        """
+        # 1 ON
+        # 2 OFF
+        enable = int(enable == False) + 1
+
+        self.write(f'disp 0,{enable}')
+        self.check_errors()
+
+    def set_lo_common_connection_relay(self, enable, lo_relay = None):
+        """ This is a command that turns on/off the connection relay
+        between the A channel LO (internal analog common)
+        and the LO (internal analog common) of the B channel.
+        """
+        # 1: Common connection relay OFF
+        # 2: Common connection relay ON
+        enable = int(enable == True) + 1
+
+        # lo_relay
+        # 1: Turn off his LO relay on all channels.
+        # 2: Turn on his LO relay on all channels.
+        # 3: Nothing is done.
+        if lo_relay is None:
+            lo_relay = 3
+        else:
+            lo_relay = int(lo_relay == True) + 1
+
+        self.write(f'ltl 0,{enable},{lo_relay}')
+        self.check_errors()
+
+    def parse_measurement(self, measurement):
+        # TODO, parse header and convert to object      
+        if ' ' in measurement:    # split, and skip header
+            measurement = measurement.split(' ')
+            return (float(measurement[1]), measurement[0])
+        else:
+            return (float(measurement), None)
+
+class AdvantestR6245(AdvantestR624X):
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            "Advantest R6245 SourceMeter",
+            **kwargs
+        )
+        self.ChA = Channel(self, 1)
+        self.ChB = Channel(self, 2)
+
+class AdvantestR6246(AdvantestR624X):
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            "Advantest R6246 SourceMeter",
+            **kwargs
+        )
+        self.ChA = Channel(self, 1)
+        self.ChB = Channel(self, 2)
+
+class Channel:
```

Comment:
We recently added the possibility to define Channels similar to Instruments (with properties...) and a convenience function for maintaining channels. See [documentation](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html#instruments-with-channels).

Reply:
Changed to ChannelCreator